File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main (int argc, string argv[])
{
    // only accepts two command line arguments
    if (argc != 2)
    {
        printf("Error: This program only accepts one command line argument.\n");
        return 1;
    }
    else
    {
        int k = atoi(argv[1]) % 26;
        
        printf("plaintext: ");
        string s = get_string();
        printf("ciphertext: ");
        
        // uses key to change plaintext to ciphertext
        for (int i = 0, n = strlen(s); i < n; i++ )
        {
            int c = s[i];
            for (int j = 0; j < k; j++)
            {
                if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122))
                {
                   if (c == 90 || c == 122)
                    c = c - 26;
                    c++; 
                }
                else 
                    break;
            }
            printf("%c", c);
        }
        
        printf("\n");
        return 0;
    }
}

File: caesar.py
import cs50
import sys

def main():
    #only accepts two command line arguments

    if len(sys.argv) != 2:
        print("Error: This program pnly accepts one command line.")
        return 1
    else:
        #converts string to int and protects wraparound
        k = int(sys.argv[1]) % 26

        print("Plaintext: ")
        s = cs50.get_string()
        print("Ciphertext: ")

        #used key to change plaintext to ciphertext

        for i in range(len(s)):
            c = ord(s[i])

            for j in range(k):

                if (c >= 65 and c <= 90) or (c >= 97 and c <= 122):
                    if c == 90 or c == 122:
                        c = c - 26

                    c += 1
                else:
                    break
            print(chr(c), end="")

    print("")
    return 0



if __name__ == "__main__":
    main()